packages/communication-react/review/communication-react.api.md

```diff
@@ -1470,8 +1472,11 @@ export interface FileUploadAdapter {
     registerFileUploads?: (fileUploads: ObservableFileUpload[]) => void;
 }
 
-// @beta (undocumented)
-export interface FileUploadEventEmitter {
+// @beta
+export class FileUploadEventEmitter {
```

Comment:
Any concerns of this must be a class? I guess we might want to follow the principle of "Separating implementation and interface", it is something like CallAdapter and AzureCommunicationCallAdapter - 
You can export an interface and use it everywhere, but implementation (the real class) could be created by another constructor or factory method(createAzureCommunicationAdapter), this leaves either us or developers more flexibility to implement them in different forms (it could be a different constructor, or it doesn't have to contain an  eventEmitter in the instance)

Reply:
@PorterNan 
created an alternative PR
https://github.com/Azure/communication-ui-library/pull/1556
I think it will address all your comments.